src/main/java/net/openhft/chronicle/network/AlwaysStartOnPrimaryConnectionStrategy.java

```diff
@@ -83,6 +85,9 @@ public ChronicleSocketChannel connect(@NotNull String name,
         else
             socketAddressSupplier.failoverToNextAddress();
 
+        if (fatalFailureMonitor == null)
+            fatalFailureMonitor = FatalFailureMonitor.NO_OP;
```

Comment:
This is listed as `@Nullable` in the constructor, then  dereferenced without first checking that it's null